### ISSUE TITLE
Puppeteer stealth mode still being detected by Datedom

### ISSUE THREAD

**Issue by rafikahmed** [berstend/puppeteer-extra#182]
Hi there, 
Leboncoin.fr still detects Puppeteer as a bot. I analyzed all the requests and I found out that they use datadom which is also anti-bot service. I tried to also inject a custom user-agent request header as others suggested by still the same issue!
Does anyone here have another workaround that I can maybe try ?
```
const userAgent = require("user-agents");
const puppeteer = require("puppeteer-extra");
const stealthPlugin = require("puppeteer-extra-plugin-stealth");
puppeteer.use(stealthPlugin());

puppeteer
  .launch({ headless: false })
  .then(async (browser) => {
    const page = await browser.newPage();
    await page.setViewport({ width: 800, height: 600 });
    await page.setUserAgent(userAgent.toString());
    await page.goto("https://www.leboncoin.fr/boutique/100524/minautor.html/");
    await page.waitFor(10000);
    await browser.close();
    console.log(`All done, check the screenshot. ✨`);
  })
  .catch((err) => console.log(err));

```
<img width="960" alt="Capture" src="https://user-images.githubusercontent.com/41229637/78940636-1ac15f00-7aae-11ea-8594-5d78bda7f0bf.PNG">

**Comment by ejames17** [comment]
@rafikahmed  im also having the same issue.

**Comment by slvDev** [comment]
Windows or macOS?
I have an issue with detecting when using macOS, but when I run a script on Windows everything working fine.
ps, it was a different site
but still, I have different behavior on windows and mac.

**Comment by rafikahmed** [comment]
Hi @slvDev,
I'm running windows, haven't tested it on Mac.

**Comment by Torpedo99** [comment]
Have you tried checking it normally using chrome and incognito? Every time I use incognito w/ chrome I seem to run in you are blocked

**Comment by ejames17** [comment]
@Torpedo99  one solution for me was to detect the js that gets loaded onto the page and abort the request as a work around.

Something like this

`    await page.setRequestInterception(true);

        page.on('request', async (request) => {
            if(request.url() === '${url}/assets/iovation-first-third.js' ){

                log(`request url is ${request.url()}: aborting`, 'warn');
                return request.abort();
            }
})
`

**Comment by Torpedo99** [comment]
yes, but in most cases, the fingerprinting script if aborted, would still result in a block.

**Comment by azerpas** [comment]
**Not puppeteer related:** 
Try to reverse-engineer their API from their App with Charles Proxy. Pretty much no security.

**Comment by Torpedo99** [comment]
@rafikahmed I have figured a way to scrape `https://www.leboncoin.fr/boutique/100524/minautor.html/`.

When running puppeteer using puppeteer stealth. Do this: 
`puppeteer.launch({ ignoreDefaultArgs: ["--enable-automation"] }).

Remember to change your proxy before you test this since datadom flags an IP address and gives out You are blocked messages once you start connecting with fresh cookies.

**Comment by ejames17** [comment]
@Torpedo99  ill have to try this method with another script. Thanks!

**Comment by leadscloud** [comment]
@Torpedo99 not work at macos,

i also have same issue in MacOS.  headless: true, can't work.  headless: false, work sucess.  i have try all setting.

**Comment by foch01** [comment]
Unresolved problem on my side I have the same problem.

![image](https://user-images.githubusercontent.com/19187357/117539151-aab33e80-b009-11eb-94dd-54beb211428b.png)

**Comment by udiudiudiudi** [comment]
have anyone recently managed to bypass datadome, I am not sure the solution outlined above still works

**Comment by stagiaire-dev** [comment]
> Problème non résolu de mon côté j'ai le même problème.
> 
> ![image](https://user-images.githubusercontent.com/19187357/117539151-aab33e80-b009-11eb-94dd-54beb211428b.png)

Have you found a solution to share?

**Comment by Unknown-Nvx** [comment]
Leboncoin.fr has very good protection against bots. For me, the solution was to control an existing web browser to make my bot interact with the UI based on the elements positions on the screen.

**Comment by udiudiudiudi** [comment]
yea laboncoin is using datadome protectoin, which is another french company,

combination of web view, saving working cookies + doing mouse movements in a bspline pattern (like normal human)  - would defeat them

**Comment by nmaisonneuve** [comment]
@udiudiudiudi 
did you manage to bypass it / can you develop a bit ? 
I am interested / DM me @nmaisonneuve

**Comment by albertlieyingadrian** [comment]
Interested in the solution to this problem still!

**Comment by duchnoun** [comment]
seem that we can't bypass now , => explanation here : https://github.com/berstend/puppeteer-extra/issues/788

**Comment by bazaned** [comment]
hello, any news?
